### PR TITLE
#Image upload functionality added #Image recognition functionality added

### PR DESCRIPTION
**Pull request by Pranav-PSU**
#Initial project scaffold for backend 
#Image upload and recognition functionality

### CODE REVIEW

File: backend/controllers/imageController.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+exports.imageStorage = multer.diskStorage({
+  destination: function (req, file, cb) {
+    console.log(process.env.NEW);
+    cb(null, process.env.DEFAULT_IMAGE_PATH);
+  },
+  filename: function (req, file, cb) {
+    cb(null, file.originalname);
+  },
+});
```
Comment (VnvsTejesh) [inline backend/controllers/imageController.js:40]: I think the code looks good I can only suggest that you could have use a good variable name instead fs.

File: backend/package.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "mindee": "^2.0.0",
+    "moment-timezone": "^0.5.37",
+    "multer": "^1.4.5-lts.1",
+    "node-schedule": "^2.1.0",
+    "path": "^0.12.7"
+  },
+  "devDependencies": {
+    "nodemon": "^2.0.20"
+  }
+}
```
Comment (VnvsTejesh) [inline backend/package.json:27]: I suggest that you could have remove unnecessary modules. Apart from that rest is good.

File: backend/routes/router.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+let uploadImage = multer({ storage: imageController.imageStorage });
+
+router.get("/scan-receipt", imageController.scanReceipt);
+router.post(
+  "/upload-receipt",
+  uploadImage.single("image"),
+  imageController.scanReceipt
+);
+
+module.exports = router;
```
Comment (VnvsTejesh) [inline backend/routes/router.js:15]: It’s a good thing that you have added the routes in a separate file.

File: backend/server.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+//TEST API
+app.get("/", function (req, res) {
+  res.status(200).send("Hello World!");
+});
+
+var port = process.env.PORT || 3000;
+app.listen(port, function () {
+  console.log("App is running on port: " + port);
+});
```
Comment (VnvsTejesh) [inline backend/server.js:51]: I would suggest that you could have removed extra cors headers and restrict access to specific origns.

**Review by VnvsTejesh — COMMENTED** [review]
ImageController.js:

I think the code looks good I can only suggest that you could have use a good variable name instead fs.

Package.JSON:

I suggest that you could have remove unnecessary modules. Apart from that rest is good.

Router.Js

It’s a good thing that you have added the route Js in a separate file.

Sever.js

I would suggest that you could have removed extra cors headers and restrict access to specific origns.

Image Service.js

I would suggest that you could have removed the empty file.

File: backend/package.json (hunk 2 above)
Comment (sarvesh2910) [reply backend/package.json:27]: Name should be something other than generic title. Better name would be 'click-n-split-backend'

File: backend/server.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+var app = express();
+const cors = require("cors");
+const path = require("path");
+const moment = require("moment-timezone");
+const dotenv = require("dotenv");
+dotenv.config();
+
+const router = require("./routes/router");
+
+app.use(bodyParser.json({ limit: "50mb" }));
```
Comment (sarvesh2910) [inline backend/server.js:12]: good catch, limiting json file size

File: backend/server.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+app.use(
+  bodyParser.urlencoded({
+    limit: "50mb",
```
Comment (sarvesh2910) [inline backend/server.js:15]: is limit getting set two times? 
line 15 and 12

**Review by sarvesh2910 — APPROVED** [review]
Added some comments. But overall looks good to me

**Pull request merged**